src/advanced_inv.cpp

```diff
@@ -1943,7 +1995,7 @@ bool advanced_inv_area::canputitems( const advanced_inv_listitem *advitem )
     switch( id ) {
         case AIM_CONTAINER:
             if( advitem != nullptr && advitem->is_item_entry() ) {
-                it = advitem->it;
+                it = (!advitem->items.empty()) ? advitem->items.front() : nullptr;
```

Comment:
Can `advitem->items` ever be empty here? The check for `advitem->is_item_entry()` should take care of that case (if the items list is empty, it should not be an item entry).